redbot/cogs/permissions/permissions.py

```diff
@@ -333,7 +332,8 @@ async def permissions_acl_getglobal(self, ctx: commands.Context):
         except discord.Forbidden:
             await ctx.send(_("I'm not allowed to DM you."))
         else:
-            await ctx.send(_("I've just sent the file to you via DM."))
+            if not isinstance(ctx.channel, discord.channel.DMChannel):
```

Comment:
It would be a bit safer to use `discord.DMChannel` name rather than `discord.channel.DMChannel` as discord.py doesn't cover the latter by the version guarantees (however unlike it is that this would get changed).

Or alternatively, you could check if `ctx.guild` is `None` instead.

Reply:
Thank you for the review, I've changed it to `discord.DMChannel`. For future PR's, what should I've done to find that out by myself? I know now that the docs tell you to use `discord.DMChannel`(should have looked that up from the start), but is that how you know that `discord.channel.DMChannel` is not covered by version guarantees? Because it's not specifically mentioned in the docs?

Reply:
Per [discord.py's version guarantees](https://discordpy.readthedocs.io/en/stable/version_guarantees.html), only documented things are part of the public API that is covered by them.

// FYI, we are doing some things in Red that are not supported by d.py but only when it's required for what we try to achieve (e.g. there's no supported way to do so) and we try to mark such things in code so that we can check them when updating dependencies.

Reply:
Got it, thank you jack